### ISSUE TITLE
Routing suggests taking a huge detour without a good reason

### ISSUE THREAD

**Issue by euf** [organicmaps/organicmaps#8850]
**Describe the issue**

Building a vehicle route.

From `34.713536, 33.266735`
To `34.709057, 33.266209`

Suggested route has a huge detour over `highway=track` with `surface=unpaved` and `smoothness=very_bad`. ETA 38 min, distance 1.7 km. 

Add stop `34.711142, 33.26732`

The route is okay now, going right through `highway=unclassified` with `surface=gravel`. ETA 1 min, distance 560 m. 

The only reason I see is the shorter route going over the road with the `vehicle=destination` access permission. It should not affect the route though since the final destination requires using this road anyways. 

**Expected behaviour**

I think OM should 
- prefer faster routes to longer ones
- prefer `surface=gravel` to `surface=unpaved` with `smoothness=very_bad`

**Screenshots**

| No stop | Added stop |
|----|----|
| ![image](https://github.com/user-attachments/assets/25288be2-8591-4d14-99e1-c0b999b71432) | ![image](https://github.com/user-attachments/assets/836cf62c-8d51-43cd-aa4c-db75313f0432) |

**System information:**
 - iOS 17.5.1, iPhone 12 mini
 - Organic Maps version: ⁦2024.07.27-6⁩
 - Map data: 23 July 2024

**Additional context**
Please add any other context or comments here that may be useful.

**Comment by muralito** [comment]
If you tap in the exclamation mark it shows a description of it. I have seen only unpaved road warning. Maybe there is other type of warning.

**Comment by euf** [comment]
Yes, there's one unpaved road segment, but two exclamation marks for some reason.

**Comment by vng** [comment]
Of course, unclassified + gravel is better than track + unpaved.

The problem is fancier here :) The end point was snapped on track road (it is a bug because unclassified is closer), so the app assumes that we should finish on track road, but unclassified is tagged as vehicle=destination, so we can't use it as a transit.

You can check it by slightly moving the end point closer to unclassified (and far from track roads) - the route will be ok.

**Comment by euf** [comment]
I'd like to note that even if the the end point would snap to the unpaved track correctly (say, 10-50 meters to the south) I'd still expect the routing go along the unpaved unclassified as long as possible before turning to the track for the last part of the drive.
